### ISSUE TITLE
Correct Method of Compiling & Running Documentation?

### ISSUE THREAD

**Issue by noddynod443** [dpethes/rerogue#12]
Hello, do you have some documentation I can closely follow for how to compile the tools via Lazarus and FreePascal, please? I tried compiling and running the de_packer/re_packer, but I get and error:

 "Project rs_repack raised exception class 'EFCreateError' with message: Unable to create file "r:\rogue_data\DATA.DAT": The system cannot find the path specified."

At address 41CEA0.

**Comment by dpethes** [comment]
Not really. Create a directory (for example "dat") next to rs_repack.exe, put the Data.dat and Data.hdr into it.
unpack with: rs_repack.exe u dat\ 
repack with: rs_repack.exe p dat\ 
Can't say what went wrong in your case. Is r: drive writable?

**Comment by noddynod443** [comment]
> Not really. Create a directory (for example "dat") next to rs_repack.exe, put the Data.dat and Data.hdr into it.
> unpack with: rs_repack.exe u dat\
> repack with: rs_repack.exe p dat\
> Can't say what went wrong in your case. Is r: drive writable?

My apologies, it was was simple error. As a friend pointed out to me, after compiling, I needed to click "run parameters" and put a C: directory location containing the dat file.

EDIT: Ok with the help of a friend, I was able to figure out how.

No insult intended, but I must say for Star Wars Sake that "I find your lack of documentation disturbing." :)

Do you know, please what are the best links to download SDL2 and ImGui (through cimgui) as I am having .dll missing errors by those names when running the modelview and such. I just am not sure what is the best place to get those.

**Comment by dpethes** [comment]
Well, some assembly required. PR-s welcome to improve the situation. This package should help with the dll-s: https://satd.sk/public/rerogue/terrain_viewer.zip

**Comment by noddynod443** [comment]
> Well, some assembly required. PR-s welcome to improve the situation. This package should help with the dll-s: https://satd.sk/public/rerogue/terrain_viewer.zip

Thank You!

Btw, I see that besides a .obj file, there is also a .raw heightmap extracted by the hmp2obj tool. No matter what software I use, the .raw heightmap doesn't open in a legible normal looking heightmap. Do you know where or how I could extract the heightmap correctly, please? I then want to convert it to a .tga file.

**Comment by dpethes** [comment]
I don't think the raw file is of any use as it is right now. I think it's broken; I can take a look at fixing it, but no promises.

**Comment by BrianAnim** [comment]
This thread helped me out a ton, I couldn't figure out what the error I was getting with repack was: "rs_repack [u|p] directory"  I thought it was asking me for a destination directory, not source.

I'm trying to get the heightmaps to import into VTOL VR and make custom missions based on rogue squadron.
https://steamcommunity.com/sharedfiles/filedetails/?id=2876350101

I was able to unpack the data files, but when I try to run the following I get an invalid filename error:

S:\Desktop\roguefiles>hmp2obj \dat\data\level\lv_0\hmp \dat\data\level\lv_0_TEXT \dat\data\level\lv_0_TEX
An unhandled exception occurred at $00000001000135B8:
EInOutError: Invalid filename
  $00000001000135B8 line 201 of rs_world.pas
  $00000001000146E6 line 396 of rs_world.pas
  $00000001000016C3 line 21 of hmp2obj.lpr
  $0000000100001836
  $000000010000FFCF
  $0000000100010BBF
  $00007FFE27097034
  $00007FFE27C026A1

If I rename lvl_0_TEX / TEXT as simply "TEX / TEXT" and put it in the same folder as the HMP it seems to work.

**Comment by dpethes** [comment]
Looks like an old version of hmp2obj and one of the paths you're passing is most likely wrong.

**Comment by BrianAnim** [comment]
Couldn't find your contact info to send you what we did with your files:
https://youtu.be/xfusENIvXX4
https://youtu.be/JgW4FbJz0eA

So thanks for publishing your tools.

**Comment by dpethes** [comment]
Looks like a fun project!
